app/models/assignment.rb

```diff
@@ -6,7 +6,10 @@ class Assignment < ApplicationRecord
 
   belongs_to :exercise
   has_one :guide, through: :exercise
-  has_many :messages, -> { where.not(submission_id: nil).order(date: :desc) }, foreign_key: :submission_id, primary_key: :submission_id, dependent: :delete_all
```

Comment:
I'm not sure we can actually remove this delete 🤔. I believe we added it explicitly because not deleting them was causing trouble. https://github.com/mumuki/mumuki-laboratory/issues/1179

Reply:
Nice catch! So please revert that single deletion and merge